media/js/base/mozilla-notification-banner.js

```diff
@@ -0,0 +1,267 @@
+        _content.appendChild(_contentContainer);
+        _contentContainer.appendChild(_heading);
+        _contentContainer.appendChild(_message);
+        _contentContainer.appendChild(_confirm);
+        _content.appendChild(_close);
+
+        return _notification;
+    };
+
+    /**
+     * Tracks impression of a notification in GA
+     * @param {Object} options - Object literal with the same format defined for init()
+     */
+    NotificationBanner.trackGAShow = function(options) {
+        var data = {
+            'data-banner-name': options.name,
+            'data-banner-impression': '1',
+            'event': 'non-interaction'
+        };
+
+        // add optional experiment specific data attributes if present.
+        if (options.experimentVariant && options.experimentName) {
+            data['data-ex-variant'] = options.experimentVariant;
+            data['data-ex-experiment'] = options.experimentName;
+            data['data-ex-present'] = 'true';
+        }
+        
+        window.dataLayer.push(data);
+    };
+
+    /**
+     * Creates, displays and tracks impression of a notification.
+     * @param {Object} options - Object literal with the same format defined for init()
+     */
+    NotificationBanner.show = function(options) {
+        var notification = NotificationBanner.create(options);
+
+        if (notification) {
+            document.body.insertBefore(notification, document.body.firstChild);
+            NotificationBanner.bind();
+            NotificationBanner.setCookie(options.id);
+            NotificationBanner.trackGAShow(options);
+        }
+    };
+
+    /**
+     * Performs basic validation for options passed to a notification.
+     * @return {Boolean}
+     */
+    NotificationBanner.validateOptions = function(options) {
+        if (typeof options.name !== 'string' ||
+            typeof options.id !== 'string' ||
+            typeof options.heading !== 'string' ||
+            typeof options.message !== 'string' ||
+            typeof options.confirm !== 'string' ||
+            typeof options.confirmLabel !== 'string' ||
+            typeof options.url !== 'string' ||
+            typeof options.close !== 'string' ||
+            typeof options.closeLabel !== 'string') {
+            return false;
+        }
+
+        return true;
+    };
+
+    /**
+     * Generic browser feature detect required for displaying a notification.
+     * @return {Boolean}
+     */
+    NotificationBanner.cutsTheMustard = function() {
+        return 'querySelector' in document &&
+               'querySelectorAll' in document &&
+               'addEventListener' in window &&
+               'createDocumentFragment' in document;
+    };
+
+    /**
+     * Initializes a new notification banner
+     * @param {Object} options - Object literal containing the following:
+     * {String} id (required) - Unique identifier used to track variations a visitor has seen e.g. 'fx-out-of-date-banner-copy1-direct-1'.
```

Comment:
Note: I'm going to follow up with proper docs for this library, but I'll do that in a follow up PR.